src/main/java/org/kainos/ea/db/AuthDao.java

```diff
@@ -0,0 +1,131 @@
+package org.kainos.ea.db;
+
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.MalformedJwtException;
+import org.apache.commons.lang3.time.DateUtils;
+import org.kainos.ea.cli.Credential;
+import org.kainos.ea.client.ActionFailedException;
+import org.kainos.ea.client.AuthenticationException;
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+
+import javax.crypto.spec.SecretKeySpec;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Base64;
+import java.util.Date;
+import java.security.Key;
+
+
+public class AuthDao {
+
+    private final Key hmacKey;
+    Argon2PasswordEncoder encoder = new Argon2PasswordEncoder(32, 64, 1, 15 * 1024, 2);
+
+    public AuthDao() throws ActionFailedException {
+        try {
+            hmacKey = new SecretKeySpec(Base64.getDecoder().decode(System.getenv("JWT_SECRET")),
+                    SignatureAlgorithm.HS256.getJcaName());
+        } catch (Exception e) {
+            throw new ActionFailedException("JWT_SECRET not set up correctly");
+        }
+    }
+
+    public void registerUser(Connection c, Credential credential) throws ActionFailedException {
+        try {
+            String encodedPassword = encoder.encode(credential.getPassword());
+
+            String insertQuery = "INSERT INTO user(username,password) VALUES(?,?)";
+            PreparedStatement st = c.prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS);
```

Comment:
This might need to go in a `try-with-resource` or be closed in a `finally` block